addons/web/static/src/js/widgets/data_export.js

```diff
@@ -49,6 +49,11 @@ var DataExport = Dialog.extend({
         this.exports = new data.DataSetSearch(this, 'ir.exports', this.record.getContext());
         this.rowIndex = 0;
         this.rowIndexLevel = 0;
+        this.isCompatibleMode = false;
+        this.groupby = this.getParent().renderer.state.groupedBy;
+        let domain = this.getParent().getActiveDomain();
+        this.domain = domain || this.record.domain;
+        this.idsToExport = domain ? false: this.getParent().getSelectedIds();
```

Comment:
my eyes are bleeding... The required information should be given at initialization

Reply:
This code was only moved  from `start` to `init` (and re-written - was even more eye-bleeding before). I didn't want to change too much. But ok to take this opportunity to clean this.
Done